flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java

```diff
@@ -43,24 +44,16 @@
 	 */
 	public static long getManagedMemorySize(Configuration configuration) {
 		long managedMemorySize;
-		String managedMemorySizeDefaultVal = TaskManagerOptions.MANAGED_MEMORY_SIZE.defaultValue();
-		if (!configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE).equals(managedMemorySizeDefaultVal)) {
-			try {
-				managedMemorySize = MemorySize.parse(
-					configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE), MEGA_BYTES).getMebiBytes();
-			} catch (IllegalArgumentException e) {
-				throw new IllegalConfigurationException("Could not read " + TaskManagerOptions.MANAGED_MEMORY_SIZE.key(), e);
-			}
-		} else {
-			managedMemorySize = Long.valueOf(managedMemorySizeDefaultVal);
+		try {
+			managedMemorySize = MemorySize.parse(
+				configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE), MEGA_BYTES).getMebiBytes();
+		} catch (IllegalArgumentException e) {
+			throw new IllegalConfigurationException("Could not read " + TaskManagerOptions.MANAGED_MEMORY_SIZE.key(), e);
 		}
-
-		checkConfigParameter(configuration.getString(
-			TaskManagerOptions.MANAGED_MEMORY_SIZE).equals(TaskManagerOptions.MANAGED_MEMORY_SIZE.defaultValue()) || managedMemorySize > 0,
+		checkConfigParameter( managedMemorySize > 0,
 			managedMemorySize, TaskManagerOptions.MANAGED_MEMORY_SIZE.key(),
 			"MemoryManager needs at least one MB of memory. " +
 				"If you leave this config parameter empty, the system automatically pick a fraction of the available memory.");
```

Comment:
I think we also need to check whether the configuration contains `MANAGED_MEMORY_SIZE` as we have just removed the default value from the `ConfigOption`. Only if `MANAGED_MEMORY_SIZE` is defined we should parse the memory size and check that it is larger than 0. If not, then we should simply return `0`.

Reply:
Thanks for your attention, MemorySize.parse() will check if the MANAGED_MEMORY_SIZE is null. Do we need to check it again?